File: backend/app/schemas/onboarding.py
"""Onboarding checklist schemas (P0.42).

Shape mirrors docs/API.md §"Onboarding". Items are returned in a
fixed order; the client is free to render them in any order but the
server's ordering matches the user's chronological progress.
"""

from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from app.schemas.common import TaxMindBooksBase

OnboardingItemKey = Literal[
    "company_created",
    "connector_installed",
    "ledgers_synced",
    "first_voucher_posted",
    "first_invoice_extracted",
]


class OnboardingItem(TaxMindBooksBase):
    key: OnboardingItemKey
    label: str
    completed: bool
    completed_at: datetime | None = None


class OnboardingChecklistResponse(TaxMindBooksBase):
    company_id: UUID
    items: list[OnboardingItem]
    completed_count: int
    total_count: int

File: mobile/src/api/onboarding.ts
/**
 * Typed bindings for /onboarding/checklist (P0.42).
 *
 * The backend uses `response_model_exclude_none=True`, so
 * `completed_at` is only present when `completed` is true. Optional
 * on the wire → optional in the type.
 */
import { api } from "./client";

export type OnboardingItemKey =
  | "company_created"
  | "connector_installed"
  | "ledgers_synced"
  | "first_voucher_posted"
  | "first_invoice_extracted";

export interface OnboardingItem {
  key: OnboardingItemKey;
  label: string;
  completed: boolean;
  completed_at?: string;
}

export interface OnboardingChecklistResponse {
  company_id: string;
  items: OnboardingItem[];
  completed_count: number;
  total_count: number;
}

export async function getOnboardingChecklist(): Promise<OnboardingChecklistResponse> {
  return api.get<OnboardingChecklistResponse>(
    "/api/v1/onboarding/checklist",
    { withCompany: true },
  );
}
